### PR TITLE
feat(cli): Fetch programs from corresponding tags

### PR DESCRIPTION
**Pull request by vadorovsky**

### CODE REVIEW

File: cli/src/psp-utils/constants.ts [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+export const SPL_NOOP_PROGRAM_TAG = "spl-noop-v0.2.0";
+export const LIGHT_MERKLE_TREE_PROGRAM_TAG = "light-merkle-tree-program-v0.3.0";
+export const LIGHT_PSP2IN2OUT_TAG = "light-psp2in2out-v0.3.0";
+export const LIGHT_PSP2IN2OUT_STORAGE_TAG = "light";
+export const LIGHT_PSP4IN4OUT_APP_STORAGE_TAG =
+  "light-psp4in4out-app-storage-v0.3.0";
+export const LIGHT_PSP10IN2OUT_TAG = "light-psp10in2out-v0.3.0";
+export const LIGHT_USER_REGISTRY_TAG = "light-user-registry-v0.3.0";
+
 export const PROGRAM_TAG = "v0.3.2";
```
Comment (ananas-block) [inline cli/src/psp-utils/constants.ts:14]: Are we still using this tag?

File: cli/src/psp-utils/constants.ts [hunk 2]
```diff
@@ -1,7 +1,18 @@
 import { RELAYER_FEE, TOKEN_ACCOUNT_FEE } from "@lightprotocol/zk.js";
 
 export const PSP_TEMPLATE_TAG = "v0.1.2";
+
+export const SPL_NOOP_PROGRAM_TAG = "spl-noop-v0.2.0";
+export const LIGHT_MERKLE_TREE_PROGRAM_TAG = "light-merkle-tree-program-v0.3.0";
+export const LIGHT_PSP2IN2OUT_TAG = "light-psp2in2out-v0.3.0";
+export const LIGHT_PSP2IN2OUT_STORAGE_TAG = "light";
```
Comment (ananas-block) [inline cli/src/psp-utils/constants.ts:8]:
```suggestion
export const LIGHT_PSP2IN2OUT_STORAGE_TAG = "light-psp2in2out-storage-v0.3.0";
```

File: cli/src/utils/initTestEnv.ts [hunk 3]
```diff
@@ -92,32 +102,42 @@ export async function start_test_validator({
...
   const LIMIT_LEDGER_SIZE = "500000000";
   const BASE_PATH = "../../bin/";
-  type Program = { id: string; name?: string; path?: string };
+  type Program = { id: string; name?: string; tag?: string; path?: string };
   const programs: Program[] = [
-    { id: "noopb9bkMVfRPU8AsbpTUg8AQkHtKwMYZiFUjNRtMmV", name: "spl_noop.so" },
+    {
+      id: "noopb9bkMVfRPU8AsbpTUg8AQkHtKwMYZiFUjNRtMmV",
+      name: "spl_noop.so",
+      tag: SPL_NOOP_PROGRAM_TAG,
```
Comment (ananas-block) [inline cli/src/utils/initTestEnv.ts:110]: It looks to me as if these tags are not used yet, but still PROGRAM_TAG.
see: https://github.com/Lightprotocol/light-protocol/pull/347/files#diff-ae098510c28e4ea810a9c0f3b0a070fccdd83301a3e2d7c424896ca2fb2474c2R167
Is this on purpose?

**Review by ananas-block — CHANGES_REQUESTED** [review]

File: cli/src/psp-utils/constants.ts (hunk 1 above)
Comment (vadorovsky) [reply cli/src/psp-utils/constants.ts]: No, sorry for missing it.

**Comment by vadorovsky** [comment]
@ananas-block Comments addresses and I added a test. PTAL

**Review by ananas-block — APPROVED** [review]
Thanks for the changes!

**Pull request merged**